### PR TITLE
Philox RNG for DPD 

### PR DESCRIPTION
**Pull request by KonradBreitsprecher**
Fixes the DPD part of #2683 

- Switched to Philox RNG
- Thermostat counter increment in common method in integrate.cpp

### CODE REVIEW

File: src/core/nonbonded_interactions/nonbonded_interaction_data.hpp [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include <dpd.hpp>
 
 #include "TabulatedPotential.hpp"
+#include "utils/Counter.hpp"
```
Comment (fweik) [inline src/core/nonbonded_interactions/nonbonded_interaction_data.hpp]: Why is this needed?

File: src/core/dpd.hpp [hunk 2]
```diff
@@ -54,5 +60,49 @@ Utils::Vector3d dpd_pair_force(Particle const *p1, Particle const *p2,
                                const IA_parameters *ia_params, double const *d,
                                double dist, double dist2);
 Utils::Vector9d dpd_stress();
+
+/** philox functiontality: increment, get/set */
+extern std::unique_ptr<Utils::Counter<uint64_t>> dpd_rng_counter;
```
Comment (fweik) [inline src/core/dpd.hpp]: I think all of the philox and counter stuff can be moved to `dpd.cpp`, as far as I can tell it's not used elsewhere.

**Review by fweik — CHANGES_REQUESTED** [review]
LGTM except for minor cleanup

File: src/core/dpd.hpp [hunk 3]
```diff
@@ -54,5 +55,17 @@ Utils::Vector3d dpd_pair_force(Particle const *p1, Particle const *p2,
                                const IA_parameters *ia_params, double const *d,
                                double dist, double dist2);
 Utils::Vector9d dpd_stress();
+
+/** philox rng */
+extern std::unique_ptr<Utils::Counter<uint64_t>> dpd_rng_counter;
```
Comment (fweik) [inline src/core/dpd.hpp]: Why do you need to expose the counter?

Comment (KonradBreitsprecher) [reply src/core/dpd.hpp]: Right, thx..

**Review by fweik — APPROVED** [review]

**Comment by fweik** [comment]
bors cancel

**Comment by jngrad** [comment]
note: e26e92f is actually a merge

**Comment by KaiSzuttor** [comment]
yeah, muscle memory...

**Comment by fweik** [comment]
bros r+

**Comment by jngrad** [comment]
bors r=fweik

**Comment by jngrad** [comment]
bors r=fweik

**Comment by fweik** [comment]
bors retry

**Pull request merged**